### PR TITLE
Most things are there and working.

### PR DESCRIPTION
**Pull request by Jaesin**
I have most things working in this branch.

### CODE REVIEW

**Comment by RoloDMonkey** [comment]
@Jaesin, I wish you had told me you were going ahead. I have been working on the same changes. I will give this a thorough review and provide feedback.

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  "homepage": "https://www.drupal.org/project/marketo_ma",
+  "minimum-stability": "dev",
+  "support": {
+    "issues": "https://www.drupal.org/project/issues/marketo_ma",
+    "source": "http://cgit.drupalcode.org/marketo_ma"
+  },
+  "repositories": [
+    {
+      "type": "vcs",
+      "url": "https://github.com/Jaesin/marketo-rest-api"
```
Comment (RoloDMonkey) [inline composer.json:16]: We will need to update URL to the real repository.

**Comment by RoloDMonkey** [comment]
@Jaesin, I understand why you are doing it but I am leery about making this module dependent on your Encryption module.

**Comment by RoloDMonkey** [comment]
@jyokum: Would you like to weigh in on this pull request?

**Comment by RoloDMonkey** [comment]
It looks like you are dropping webform support. This makes sense for now.

File: composer.json (hunk 1 above)
Comment (Jaesin) [reply composer.json:16]: The repositories section can be removed once an 8.x-1.x branch gets pushed to d.o.

**Comment by Jaesin** [comment]
Yeah, the encryption module was made for this module specifically. The need is simple encryption so when api credentials get written to config they aren't publicly readable by a improperly configured server. It would be nice if there was a simple encryption mechanism in core and I wrote this one hoping to get it into a minor release.

Some people are using contact storage until something better comes along.

File: composer.json (hunk 1 above)
Comment (Jaesin) [reply composer.json:16]: Ah, my mistake. that is there until my pull request is accepted: https://github.com/dchesterton/marketo-rest-api/pull/34

**Comment by Jaesin** [comment]
There are lots of tests but they rely on api credentials in environment variable. I have mine set up in PHPStorm and all tests are passing.

Required environment variables to get all tests passing. 

marketo_ma_munchkin_account_id
marketo_ma_rest_client_id
marketo_ma_rest_client_secret
marketo_ma_munchkin_api_private_key

This can be set up using [travis encryption](https://docs.travis-ci.com/user/environment-variables/#Defining-encrypted-variables-in-.travis.yml) but someone would need marketo sandbox credentials that they are comfortable with using.

**Comment by jyokum** [comment]
Starting to review

**Comment by mccrodp** [comment]
> It looks like you are dropping webform support. This makes sense for now.

Would be great to integrate with EntityForm and/or YML form instead <- Nice roadmap goal :P Great job w/ this, won't have chance to review properly, but from what I saw before, looks very thorough 👍

**Comment by Jaesin** [comment]
I wanted to make adding additional integrations a trivial task but didn't have time to optimize for that.

It's a target of opportunity for sure.

**Comment by Jaesin** [comment]
This last commit fixes a config issue where if you uninstall mma_user, the config wouldn't uninstall. Then when you try to re-install, you get an error. It adds some refactoring on where the actual config names are stored.

**Comment by Jaesin** [comment]
Ah, because they are two different values.

``` php
const MMA_USER_CONFIG_NAME   = 'mma_user.settings';
const MARKETO_MA_CONFIG_NAME = 'marketo_ma.settings';
```

**Comment by RoloDMonkey** [comment]
Right. I guess I was reading too fast. :)

**Comment by jyokum** [comment]
I'm new to D8 and struggling a bit to get real testing going but here are a few thoughts from what I have observed so far.

encryption - Agree with @RoloDMonkey that the dependency to the other module is undesirable. If you have access to the server it's pretty easy to get at the real keys even when encrypted.
`drush ev "var_dump(\Drupal::service('marketo_ma.api_client'))"`

Would like to understand the rationale for renaming the submodules? marketo_ma_user -> mma_user

Prefer to leave default tracking method as Munchkin given it's the most straightforward setup and is all that's really needed if you're not going to also enable lead sync through one of the other submodules.

**Comment by Jaesin** [comment]
About renaming sub-modules. I didn't think the machine name was so important since the you can't upgrade a drupal 7 site but I'm pretty agnostic to their names.

About encryption, It's a common practice export config to files. The idea with encryption isn't to protect from someone that has server access, They can just log in and see the credentials on the configuration page if they have that. The idea is to protect against public access because of a poorly configured server. If a site doesn't protect `sites/default/config` properly, the yaml files could be publicly exposed. 

We work with allot of enterprise clients but we don't have control over servers in many cases. So we have to do our part (due diligence) to protect sensitive data for our clients.

All that said, the encryption trait is only 33 lines of code, we could easily duplicate it inside the `marketo_ma` module until a sufficient encryption trait makes it's way into core.

Default tracking set to munchkin makes since.

**Comment by Jaesin** [comment]
I think most minor things can be sorted in followup issues but I added both RoloDMonkey and  jyokum as collaborators to https://github.com/Jaesin/marketo_ma. Feel free to push any changes you guys need to make before accepting this pull request.

I still have about a week or so before I need to really implement for a client.

**Comment by RoloDMonkey** [comment]
If I understand correctly, it looks like we need a "go"/"no go" on two things before we can do a dev release and put this up on drupal.org:
1. Dependency on the encryption module.
2. Name of the sub-modules.

Which way are we going? I can do the work once I know what the decisions are.

**Comment by jyokum** [comment]
Let's keep the encryption module in and rename the submodules to follow pattern marketo_ma_XYZ

**Comment by RoloDMonkey** [comment]
Sounds good.

@Jaesin, Do you want to do the refactoring of the module names? I have some time today, if you need help.

**Comment by Jaesin** [comment]
@RoloDMonkey I am a bit busy today. If you have time it's great to get the help. Just have to make sure to rename `mma_user.settings.yml` back to `marketo_ma_user.settings.yml`.

**Comment by RoloDMonkey** [comment]
I will start on the refactor now.

**Comment by RoloDMonkey** [comment]
@Jaesin, refactoring is complete. You will definitely want to review and re-run the tests.

**Comment by Jaesin** [comment]
After one small fix everything looks good from the tests.

**Comment by RoloDMonkey** [comment]
@jyokum: I think this is ready to merge and become the new 8.x branch on drupal.org. Please let me know immediately if you have any questions.

**Comment by jyokum** [comment]
Great. I think I'd prefer to make this the 8.x-2.x release to correspond with 7.x-2.x being the move to REST API. Any objection?

**Comment by RoloDMonkey** [comment]
@jyokum: No objections: https://www.drupal.org/node/2610556#comment-11620597

:)

**Comment by Jaesin** [comment]
@RoloDMonkey Thanks for the help.

**Pull request merged**